src/include/rbd/librbd.hpp

```diff
@@ -176,6 +176,17 @@ class CEPH_RBD_API ImageOptions {
   rbd_image_options_t opts;
 };
 
+class CEPH_RBD_API UpdateWatchCtx {
+public:
+  virtual ~UpdateWatchCtx() {}
+  /**
+   * Callback activated when we receive a notify event.
+   */
+  virtual void handle_notify() = 0;
+
+  virtual void handle_unregister() {}
```

Comment:
Instead of using a dummy virtual function to handle the C API deletion problem, could you just have the C API return a pointer to the allocated ```C_UpdateWatchCB``` object via the handle and for unwatch you could cast the handle back to ```C_UpdateWatchCB```, extract the real handle, unwatch, and delete.